src/lib/RoomConnection.ts

```diff
@@ -644,15 +692,31 @@ export default class RoomConnection extends TypedEventTarget {
         });
     }
 
-    private _handleStreamAdded({ clientId, stream, streamId }: RtcStreamAddedPayload) {
+    private _handleStreamAdded({ clientId, stream, streamId, streamType }: RtcStreamAddedPayload) {
         const remoteParticipant = this.remoteParticipants.find((p) => p.id === clientId);
         if (!remoteParticipant) {
             this.logger.log("WARN: Could not find participant for incoming stream");
             return;
         }
 
+        const remoteParticipantStream = remoteParticipant.streams.find((s) => s.id === streamId);
+
+        if (
+            (remoteParticipant.stream && remoteParticipant.stream.id === streamId) ||
+            (!remoteParticipant.stream && streamType === "webcam") ||
+            (!remoteParticipant.stream &&
```

Comment:
It seems there are cases where these complex if statement returns false for real participants. 😕 

Actually I couldn't make this work, my pwa client was always rendered as a screenshare, however it appeared in the participant list as well 🤯 


<img width="1505" alt="Screenshot 2023-09-06 at 12 07 42" src="https://github.com/whereby/browser-sdk/assets/1159931/8f001cdc-bf3d-409d-a6b8-81c4fb82e430">

related code in my test app:

```jsx
<h3>Participants</h3>
<div>
    {remoteParticipants.map((r) => (
        <div key={r.id}>
            <VideoView
                style={{ width: remoteVideoWidth }}
                stream={r.stream}
            />
            <p>{r.displayName}</p>
        </div>
    ))}
</div>
{screenshares?.length > 0 && (
    <div>
        <h3>Screenshares</h3>
        {screenshares.map((s) => (
            <div key={s.id || s.participantId}>
                <VideoView
                    style={{ width: remoteVideoWidth }}
                    stream={s.stream}
                />
            </div>
        ))}
    </div>
)}

```

Reply:
This is more or less exactly what we have already in the pwa so I'm surprised that this doesn't work for you. Can't reproduce either - it works on my side. Did you do a `yarn install` before building the SDK? maybe it's that.

Reply:
Hm interesting, it was a cache issue on my end probably, clearing everything and starting again made it work, sorry for the noise then.